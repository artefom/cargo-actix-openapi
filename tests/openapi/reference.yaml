openapi: '3.0.3'
info:
  version: '1.0'
  title: 'Hello World API'
paths:
  /hello/{user}:
    parameters:
      - $ref: '#/components/parameters/User'
    get:
      operationId: greet_user
      summary: Returns a greeting to the user!
      description: Some detailed description
      responses:
        200:
          description: Returns the greeting.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/HelloUserResponse'
components:
  parameters:
    User:
      name: user
      in: path
      required: true
      description: The name of the user to greet.
      schema:
        type: string
  schemas:
    HelloUserResponse:
      type: string
