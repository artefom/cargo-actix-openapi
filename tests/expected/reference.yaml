api:
  definitions:
    DOCS_OPENAPI:
      data: !StaticStr
        path: static/openapi.yaml
    DOCS_HTML:
      data: !StaticStr
        path: static/docs.html
    openapi:
      data: !StaticStringPath
        data: DOCS_OPENAPI
    docs:
      data: !StaticHtmlPath
        data: DOCS_HTML
    to_docs:
      data: !Redirect
        target: v1/docs
    GreetUserPath:
      data: !Struct
        doc: null
        properties:
        - name: user
          rename: user
          default: null
          type_: String
          doc: The name of the user to greet.
  operations:
    greet_user:
      doc: Returns a greeting to the user!
      param_path: web::Path<GreetUserPath>
      param_query: null
      param_body: null
      response: web::Json<String>
  paths:
  - operation: greet_user
    path: /hello/{user}
    method: Get
  - operation: greet_user
    path: /v1/hello/{user}
    method: Get
  static_services:
  - method: Get
    path: /openapi.yaml
    data: openapi
  - method: Get
    path: /docs
    data: docs
  - method: Get
    path: /v1
    data: to_docs
  - method: Get
    path: /v1/openapi.yaml
    data: openapi
  - method: Get
    path: /v1/docs
    data: docs
